core/src/main/java/com/linecorp/armeria/server/healthcheck/CpuHealthChecker.java

```diff
@@ -0,0 +1,174 @@
+        currentSystemCpuUsageSupplier = () -> invoke(systemCpuUsage);
+        currentProcessCpuUsageSupplier = () -> invoke(processCpuUsage);
+    }
+
+    private final double targetCpuUsage;
+
+    private final DoubleSupplier systemCpuUsageSupplier;
+
+    private final DoubleSupplier processCpuUsageSupplier;
+
+    private final double targetProcessCpuLoad;
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param cpuUsageThreshold the target cpu usage
+     * @param processCpuLoadThreshold the target process cpu usage
+     */
+    private CpuHealthChecker(double cpuUsageThreshold, double processCpuLoadThreshold) {
+        this(cpuUsageThreshold, processCpuLoadThreshold,
+                currentSystemCpuUsageSupplier, currentProcessCpuUsageSupplier);
+    }
+
+    private CpuHealthChecker(double cpuUsageThreshold, double processCpuLoadThreshold,
+                             DoubleSupplier systemCpuUsageSupplier, DoubleSupplier processCpuUsageSupplier) {
+        this.targetCpuUsage = cpuUsageThreshold;
+        this.targetProcessCpuLoad = processCpuLoadThreshold;
+        this.systemCpuUsageSupplier = systemCpuUsageSupplier;
+        this.processCpuUsageSupplier = processCpuUsageSupplier;
+    }
+
+    private static double invoke(@Nullable MethodHandle mh) {
+        if (mh == null) {
+            return Double.NaN;
+        }
+
+        try {
+            return (double) mh.invoke(operatingSystemBean);
+        } catch (Throwable e) {
+            return Double.NaN;
+        }
+    }
+
+    @Nullable
+    @SuppressWarnings("ReturnValueIgnored")
+    private static MethodHandle detectMethod(final String name) {
+        if (operatingSystemBeanClass == null) {
+            return null;
+        }
+        try {
+            // ensure the Bean we have is actually an instance of the interface
+            operatingSystemBeanClass.cast(operatingSystemBean);
+            final Method method = operatingSystemBeanClass.getMethod(name);
+            return LOOKUP.unreflect(method);
+        } catch (ClassCastException | NoSuchMethodException | SecurityException | IllegalAccessException e) {
+            logger.warn(
+                    "Failed to detect method {}.{} for {}",
+                    operatingSystemBeanClass.getSimpleName(),
+                    name,
+                    CpuHealthChecker.class.getSimpleName(),
+                    e
+            );
+            return null;
+        }
+    }
+
+    @Nullable
+    private static Class<?> getFirstClassFound(final List<String> classNames) {
+        for (String className : classNames) {
+            try {
+                return Class.forName(className, false, CpuHealthChecker.class.getClassLoader());
+            } catch (ClassNotFoundException ignore) {
+            }
+        }
+        return null;
+    }
+
+    public static CpuHealthChecker of(double targetCpuUsage, double targetProcessCpuLoad) {
```

Comment:
@jrhee17 If I make this change, it will cause modifications to other files that implement this interface.
I think it would be better to address this in a new issue. If you have any other ideas, please let me know 🙂